cmsplugin_articles_ai/models.py

```diff
@@ -0,0 +1,286 @@
+# -*- coding: utf-8 -*-
+try:
+    from urllib.parse import urlencode  # Python 3
+except ImportError:
+    from urllib import urlencode  # Python 2
+
+from cms.models import CMSPlugin
+from django.contrib.auth.models import User
+from django.core.urlresolvers import reverse
+from django.db import models
+from django.db.models import Count, Q
+from django.utils import timezone
+from django.utils.encoding import python_2_unicode_compatible
+from django.utils.translation import ugettext_lazy as _
+from djangocms_text_ckeditor.fields import HTMLField
+from enumfields import Enum, EnumIntegerField
+from filer.fields.file import FilerFileField
+from filer.fields.image import FilerImageField
+
+
+class TagQuerySet(models.QuerySet):
+
+    @property
+    def as_url_encoded(self):
+        """
+        Return queryset as URL encoded string that is compatible with
+        the article views.
+        """
+        return urlencode({
+            "filter_tags": self.values_list("pk", flat=True)
+        })
+
+
+@python_2_unicode_compatible
+class Tag(models.Model):
```

Comment:
Maybe separate the two models-and-managers into distinct files in a models package?